core/modules/map.rs

```diff
@@ -770,50 +821,25 @@ impl ModuleMap {
   fn is_registered(
     &self,
     specifier: impl AsRef<str>,
-    asserted_module_type: AssertedModuleType,
+    asserted_module_type: impl AsRef<AssertedModuleType>,
   ) -> bool {
-    if let Some(id) = self.get_id(specifier.as_ref(), asserted_module_type) {
-      return asserted_module_type
-        == self.data.borrow().info.get(id).unwrap().module_type.into();
-    }
-
-    false
-  }
-
-  pub(crate) fn by_name<T>(
-    &self,
-    asserted_module_type: AssertedModuleType,
-    f: impl FnOnce(&HashMap<ModuleName, SymbolicModule>) -> T,
-  ) -> T {
-    match asserted_module_type {
-      AssertedModuleType::Json => f(&self.data.borrow().by_name_json),
-      AssertedModuleType::JavaScriptOrWasm => f(&self.data.borrow().by_name_js),
-    }
-  }
-
-  pub(crate) fn by_name_mut<T>(
-    &self,
-    asserted_module_type: AssertedModuleType,
-    f: impl FnOnce(&mut HashMap<ModuleName, SymbolicModule>) -> T,
-  ) -> T {
-    match asserted_module_type {
-      AssertedModuleType::Json => f(&mut self.data.borrow_mut().by_name_json),
-      AssertedModuleType::JavaScriptOrWasm => {
-        f(&mut self.data.borrow_mut().by_name_js)
-      }
-    }
+    self
```

Comment:
This was overtesting -- if it's in `ModuleInfo` with the given ID, the type is correct.

Reply:
Do we have a unit test for that? If not could you add one?

Reply:
Unless I'm mistaken, all the tests technically cover this functionality because it's just not possible for the type not to match. The modules of different types are all independent, so the module returned by `get_id` _must_ have the same module type. I suspect this code was leftover from when I split the maps up ages ago and just wasn't cleaned up.